Library/Homebrew/tap.rb

```diff
@@ -362,6 +362,29 @@ def link_completions_and_manpages
     end
   end
 
+  def fix_remote_configuration
+    return unless remote.include? "github.com"
+
+    current_upstream_head = Utils.popen_read("git", "-C", path, "symbolic-ref", "refs/remotes/origin/HEAD")
+                                 .sub(%r{^refs/remotes/origin/}, "").chomp
+    Utils.popen_read("git", "-C", path, "ls-remote", "--heads", "--exit-code", "origin", current_upstream_head)
+    return if $CHILD_STATUS.success?
+
+    ohai "#{name}: Unable to find the remote branch (#{current_upstream_head})", <<~EOS
+      Searching for a new origin/HEAD...
+    EOS
+
+    repo_data = GitHub.repository(*full_name.split("/"))
+    new_upstream_head = repo_data["default_branch"]
+
+    Utils.popen_read "git", "-C", path, "fetch", "origin", new_upstream_head
+    Utils.popen_read "git", "-C", path, "remote", "set-head", "origin", new_upstream_head
+    Utils.popen_read "git", "-C", path, "branch", "-m", current_upstream_head, new_upstream_head
+    Utils.popen_read "git", "-C", path, "branch", "-u", "origin/#{new_upstream_head}", new_upstream_head
```

Comment:
Can we use `popen_git` or new functions in `git_repository,rb` to handle some of this?